node/service/src/lib.rs

```diff
@@ -860,17 +857,25 @@ where
 		config.network.extra_sets.extend(peer_sets_info(is_authority));
 	}
 
-	let (pov_req_receiver, cfg) = IncomingRequest::get_config_receiver();
+	let fork_id = config.chain_spec.fork_id().map(ToOwned::to_owned);
+
+	let (pov_req_receiver, cfg) =
+		IncomingRequest::get_config_receiver(&genesis_hash, fork_id.as_deref());
```

Comment:
Nit: I would actually like it better if we used the new `ReqProtocolNames` as the source of truth. So we only pass in genesis and fork id once into `ReqProtocolNames::new()` and then use that `ReqProtocolNames` everywhere.

Reply:
Very good idea!

And then we could also pass this directly to the overseer below and don't need to pass `fork_id`!

Reply:
Done! About to merge.